grammar.js

```diff
@@ -26,6 +26,10 @@ const PREC = {
   COMPLEMENT: 85,
 };
 
+const IDENTIFIER_CHARS = /[^\s:;`"'@$#.,|^&<=>+\-*/\\%?!~()\[\]{}]*/;
+const LOWER_ALPHA_CHAR = /[^\sA-Z0-9:;`"'@$#.,|^&<=>+\-*/\\%?!~()\[\]{}]/;
+const ALPHA_CHAR = /[^\s0-9:;`"'@$#.,|^&<=>+\-*/\\%?!~()\[\]{}]/;
```

Comment:
Could `LOWER_ALPHA_CHAR` and `ALPHA_CHAR` be simplified back to something like `[a-z_]` and `[A-Za-z_]`, or do they need this more general structure?

Reply:
They need to allow for things like an ivar named `@åäö`.

Reply:
Oh that makes sense; I didn't realize that even the first letter could be any character like that.